### PR TITLE
Live Preview: Fix the broken z-index of the introduction modal

### PR DESCRIPTION
**Pull request by okmttdhr**
Originally raised in: p7DVsv-iTp-p2#comment-48006
Related to: https://github.com/Automattic/wp-calypso/pull/82666

## Proposed Changes

This PR fixes the broken z-index of the introduction modal in Live Preview (Preview & Customize).

| before | after |
|--------|--------|
| Screenshot 2023-10-19 at 15 35 43 | Screenshot 2023-10-19 at 15 32 33 | 

## Testing Instructions

1. Patch this change to your sandbox: `install-plugin.sh editing-toolkit fix/live-preview-modal-sp`.
1. Sandbox YOUR SITE.
1. Simulate the mobile device in your Chrome DevTools.
1. Now go to Appearance -> Themes.
1. Choose a Free theme and click "Preview & Customize".
1. Verify that the modal is shown in front of every element.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/live-preview-modal/style.scss
Comment (okmttdhr) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/live-preview-modal/style.scss:22]: Let me know if we have moer appropriate value.

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/live-preview-modal/style.scss
Comment (okmttdhr) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/live-preview-modal/style.scss]: `z-index` was defined in https://github.com/Automattic/wp-calypso/blob/3292129/packages/components/src/dialog/style.scss#L16, 

I set this manually, as live-preview-modal is out of Calypso, and there is no way to use https://github.com/Automattic/wp-calypso/blob/3292129/client/assets/stylesheets/shared/functions/_z-index.scss#L153 (CMIIW).

**Review by fushar — APPROVED** [review]
Nice 👍 verified, thanks!

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/live-preview-modal/style.scss
Comment (miksansegundo) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/live-preview-modal/style.scss:1]: Can we reduce the padding of `.dialog__content.live-preview-modal` on mobile? Something like `padding: 48px 32px 28px;`

|Before|After|
|--|--|
|Screenshot 2566-10-19 at 17 02 02|Screenshot 2566-10-19 at 17 01 44|

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/live-preview-modal/style.scss
Comment (miksansegundo) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/live-preview-modal/style.scss]: Other components in `wpcom-block-editor-nux` are using `Modal` from `@wordpress/components` and don't have this issue.

I think we should use [Modal](https://github.com/WordPress/gutenberg/tree/7b8cb73fdfac5621a44e72f2aee3a4929b400023/packages/components/src/modal) rather than `Dialog`. cc: @fushar 

However, Dialog works fine in this PR with the z-index fixed.

I think it's safer to use `100000` because that is the z-index used for `.components-modal__screen-overlay`. See https://github.com/WordPress/gutenberg/blob/7b8cb73fdfac5621a44e72f2aee3a4929b400023/packages/base-styles/_z-index.scss#L116

```suggestion
	// Dialog sets `z-index: z-index("root", ".dialog__backdrop")`,
	// but this modal is used outside of Calypso, so we need to set it manually.
	z-index: 100000;
```

**Review by miksansegundo — APPROVED** [review]
🚀 LGTM if we don't want to refactor it to use `Modal` from `@wordpress/components`.

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/live-preview-modal/style.scss
Comment (fushar) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/live-preview-modal/style.scss]: We can refactor, but I'd personally prefer we allocate our time to work on the smart modal later which will be done outside `wpcom-block-editor-nux` anyway (e.g. on `jetpack-mu-plugins`). Up to @okmttdhr  😄

Comment (okmttdhr) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/live-preview-modal/style.scss]: Thank you for your comment, @miksansegundo @fushar!

>Other components in wpcom-block-editor-nux are using Modal from @wordpress/components and don't have this issue.
I think we should use [Modal](https://github.com/WordPress/gutenberg/tree/7b8cb73fdfac5621a44e72f2aee3a4929b400023/packages/components/src/modal) rather than Dialog. 

This sounds make sense. It'd be nice to use @wordpress/components in this kind of situation.
For now, I would not change it to use Modal, as this modal itself should be considered as an interim solution. https://github.com/Automattic/wp-calypso/issues/82187 But I'll add a comment about this!

>I think it's safer to use 100000 because that is the z-index used for .components-modal__screen-overlay. See [WordPress/gutenberg@7b8cb73/packages/base-styles/_z-index.scss#L116](https://github.com/WordPress/gutenberg/blob/7b8cb73fdfac5621a44e72f2aee3a4929b400023/packages/base-styles/_z-index.scss#L116)

I'll address this 👍

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/live-preview-modal/style.scss
Comment (fushar) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/live-preview-modal/style.scss]: Sorry, can we revert this? It looks bad on Desktop, where most of live-previews will happen:

image

I see there are two solutions:

1. Define this padding on a media breakpoint, or
2. Just not do it? Because I see all other modals are having the same issues 😅 e.g.

image

Comment (okmttdhr) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/live-preview-modal/style.scss]: I'll use a breakpoint!

**Review by fushar — APPROVED** [review]
🚢

**Pull request merged**
